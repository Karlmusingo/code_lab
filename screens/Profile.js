import React, { Component } from "react";
import {
  StyleSheet,
  View,
  Text,
  TouchableHighlight,
  ImageBackground,
  Image,
  Dimensions,
  ScrollView,
  SafeAreaView
} from "react-native";
import { Icon } from "react-native-elements";
import ProfileHeader from "../components/ProfileHeader";

class Profile extends Component {
  state = {};

  render() {
    const { navigation } = this.props;
    const dev = navigation.getParam("dev", "NO-ID");

    return (
      <View>
        <ProfileHeader
          login={dev.login}
          goBack={() => navigation.goBack(null)}
        />
        <ScrollView>
          <View style={{}}>
            <ImageBackground
              source={{
                uri: dev.avatarUrl
              }}
              style={{ height: 540, width: Dimensions.width }}
            >
              <View
                style={{
                  height: "100%",
                  width: "100%",
                  backgroundColor: "rgba(0, 0, 0, 0.75)"
                }}
              >
                <View
                  style={{
                    display: "flex",
                    flexDirection: "row",
                    justifyContent: "space-between",
                    width: "100%",
                    position: "absolute",
                    top: 370,
                    alignItems: "flex-end",
                    paddingRight: 20,
                    paddingLeft: 20
                  }}
                >
                  <Text style={{ width: 39 }} />
                  <Image
                    source={{
                      uri: dev.avatarUrl
                    }}
                    style={{ height: 140, width: 140, borderRadius: 70 }}
                  />
                  <TouchableHighlight
                    onPress={() =>
                      navigation.navigate("Web", { login: dev.login })
                    }
                  >
                    <Icon
                      name="share-square"
                      size={45}
                      color="#fff"
                      type="font-awesome"
                    />
                  </TouchableHighlight>
                </View>
              </View>
            </ImageBackground>
          </View>
          <View
            style={{
              display: "flex",
              alignItems: "center",
              padding: 20
            }}
          >
            <Text style={{ fontWeight: "700", fontSize: 25 }}>{dev.name}</Text>
            <Text style={{ fontWeight: "400", fontSize: 20, color: "#47525E" }}>
              @{dev.login}
            </Text>
            <Text
              style={{
                fontWeight: "400",
                fontSize: 15,
                textAlign: "center",
                paddingTop: 20,
                paddingBottom: 20
              }}
            >
              {dev.bio}
            </Text>
            <View
              style={{
                display: "flex",
                flexDirection: "row",
                justifyContent: "space-around",
                width: "100%"
              }}
            >
              <View>
                <Text style={styles.text}>{dev.followers.totalCount}</Text>
                <Text style={styles.text}>Followers</Text>
              </View>
              <View>
                <Text style={styles.text}>{dev.following.totalCount}</Text>
                <Text style={styles.text}>Followings</Text>
              </View>
              <View>
                <Text style={styles.text}>
                  {dev.starredRepositories.totalCount}
                </Text>
                <Text style={styles.text}>Stars</Text>
              </View>
            </View>
            <View style={{ marginTop: 30 }}>
              <Text
                style={{ fontWeight: "700", fontSize: 35, textAlign: "center" }}
              >
                {dev.repositories.totalCount}
              </Text>
              <Text
                style={{ fontWeight: "700", fontSize: 20, textAlign: "center" }}
              >
                Repositories
              </Text>
            </View>
          </View>
        </ScrollView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  text: {
    fontWeight: "400",
    fontSize: 20,
    color: "#47525E",
    textAlign: "center"
  }
});

export default Profile;
