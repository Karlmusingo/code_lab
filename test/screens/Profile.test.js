import React from "react";
import { shallow, mount } from "enzyme";
import { MockedProvider } from "react-apollo/test-utils";
import renderer from "react-test-renderer";
import Profile from "../../screens/Profile";

describe("Render the profile component", () => {
  const props = {
    navigation: {
      getParam: jest.fn(() => ({
        avatarUrl: "avatar.url",
        name: "name",
        bio: "",
        avatarUrl: "",
        createdAt: "",
        starredRepositories: {
          totalCount: 3
        },
        repositories: {
          totalCount: 3
        },
        followers: {
          totalCount: 3
        },
        following: {
          totalCount: 3
        }
      })),
      navigate: jest.fn()
    }
  };

  it("should render the profile component", async () => {
    const component = shallow(<Profile {...props} />);
    component.find("TouchableHighlight").simulate("press");
    expect(component).toMatchSnapshot();
  });
});
