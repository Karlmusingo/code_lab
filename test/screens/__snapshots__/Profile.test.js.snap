// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the profile component should render the profile component 1`] = `
<View>
  <ProfileHeader
    goBack={[Function]}
  />
  <ScrollViewMock>
    <View
      style={Object {}}
    >
      <ImageBackground
        source={
          Object {
            "uri": "",
          }
        }
        style={
          Object {
            "height": 540,
            "width": undefined,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.75)",
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingLeft": 20,
                "paddingRight": 20,
                "position": "absolute",
                "top": 370,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "width": 39,
                }
              }
            />
            <Image
              source={
                Object {
                  "uri": "",
                }
              }
              style={
                Object {
                  "borderRadius": 70,
                  "height": 140,
                  "width": 140,
                }
              }
            />
            <TouchableHighlight
              activeOpacity={0.85}
              delayPressOut={100}
              onPress={[Function]}
              underlayColor="black"
            >
              <Themed.Icon
                color="#fff"
                name="share-square"
                size={45}
                type="font-awesome"
              />
            </TouchableHighlight>
          </View>
        </View>
      </ImageBackground>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "padding": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 25,
            "fontWeight": "700",
          }
        }
      >
        name
      </Text>
      <Text
        style={
          Object {
            "color": "#47525E",
            "fontSize": 20,
            "fontWeight": "400",
          }
        }
      >
        @
      </Text>
      <Text
        style={
          Object {
            "fontSize": 15,
            "fontWeight": "400",
            "paddingBottom": 20,
            "paddingTop": 20,
            "textAlign": "center",
          }
        }
      />
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-around",
            "width": "100%",
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#47525E",
                "fontSize": 20,
                "fontWeight": "400",
                "textAlign": "center",
              }
            }
          >
            3
          </Text>
          <Text
            style={
              Object {
                "color": "#47525E",
                "fontSize": 20,
                "fontWeight": "400",
                "textAlign": "center",
              }
            }
          >
            Followers
          </Text>
        </View>
        <View>
          <Text
            style={
              Object {
                "color": "#47525E",
                "fontSize": 20,
                "fontWeight": "400",
                "textAlign": "center",
              }
            }
          >
            3
          </Text>
          <Text
            style={
              Object {
                "color": "#47525E",
                "fontSize": 20,
                "fontWeight": "400",
                "textAlign": "center",
              }
            }
          >
            Followings
          </Text>
        </View>
        <View>
          <Text
            style={
              Object {
                "color": "#47525E",
                "fontSize": 20,
                "fontWeight": "400",
                "textAlign": "center",
              }
            }
          >
            3
          </Text>
          <Text
            style={
              Object {
                "color": "#47525E",
                "fontSize": 20,
                "fontWeight": "400",
                "textAlign": "center",
              }
            }
          >
            Stars
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 30,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 35,
              "fontWeight": "700",
              "textAlign": "center",
            }
          }
        >
          3
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "700",
              "textAlign": "center",
            }
          }
        >
          Repositories
        </Text>
      </View>
    </View>
  </ScrollViewMock>
</View>
`;
